apps/webapp/app/services/db/pgListen.server.ts

```diff
@@ -0,0 +1,48 @@
+import { logger } from "~/services/logger.server";
+import { Logger } from "@trigger.dev/core";
+import type { PoolClient } from "pg";
+
+export class PgListenService {
+  #poolClient: PoolClient;
+  #logger: Logger;
+  #loggerNamespace: string;
+
+  constructor(poolClient: PoolClient, loggerNamespace?: string, loggerInstance?: Logger) {
+    this.#poolClient = poolClient;
+    this.#logger = loggerInstance ?? logger;
+    this.#loggerNamespace = loggerNamespace ?? "";
+  }
+
+  public async call(channelName: string, callback: (payload: string) => Promise<void>) {
+    this.#logDebug("Registering notification handler", { channelName });
+
+    const isValidChannel = channelName.match(/^[a-zA-Z0-9:-_]+$/);
+
+    if (!isValidChannel) {
+      throw new Error(`Invalid channel name: ${channelName}`);
+    }
+
+    this.#poolClient.query(`LISTEN "${channelName}"`).then(null, (error) => {
```

Comment:
`channelName` gets validated above to prevent SQL injections. Don't think we can pass it as a param to `query()`, which would prevent having to do this.